lib/dal-materialized-views/src/luminork/change_set_review.rs

```diff
@@ -0,0 +1,265 @@
+fn to_attribute_diff_trees(
+    attribute_diffs: &[(String, AttributeDiff)],
+) -> crate::Result<Vec<AttributeDiffTree>> {
+    let mut result = Vec::new();
+
+    for (path, diff) in attribute_diffs {
+        // Filter out uninteresting diffs
+        if !should_include_diff(path, diff) {
+            continue;
+        }
+
+        result.push(AttributeDiffTree {
+            path: path.clone(),
+            diff: diff.clone(),
+        });
+    }
+
+    // Sort by path for consistent ordering
+    result.sort_by(|a, b| a.path.cmp(&b.path));
+
+    Ok(result)
+}
+
+/// Filters out uninteresting diffs (matching frontend logic)
+fn should_include_diff(path: &str, diff: &AttributeDiff) -> bool {
+    use si_frontend_mv_types::component::component_diff::AttributeSourceAndValue;
+
+    // Filter out internal SI attributes that aren't useful for review
+    if path == "/si/type" || path == "/si/color" {
+        return false;
+    }
+
+    let (old, new): (
+        Option<&AttributeSourceAndValue>,
+        Option<&AttributeSourceAndValue>,
+    ) = match diff {
+        AttributeDiff::Added { new } => (None, Some(new)),
+        AttributeDiff::Removed { old } => (Some(old), None),
+        AttributeDiff::Modified { old, new } => (Some(old), Some(new)),
+    };
+
+    // If old and new are identical, skip (can happen on upgrades)
+    if let (Some(old_val), Some(new_val)) = (old, new) {
+        if old_val == new_val {
+            return false;
+        }
+    }
+
+    // For added/removed values, filter out uninteresting defaults
+    if old.is_none() || new.is_none() {
+        let source_and_value = old.or(new).unwrap();
+
+        // Don't show uninteresting default values from schema
+        if source_and_value.source.from_schema.unwrap_or(false) {
+            if let Some(value) = &source_and_value.value {
+                // Filter empty objects/arrays, null, empty string, zero
+                if value.is_null()
+                    || (value.is_object() && value.as_object().map_or(false, |o| o.is_empty()))
+                    || (value.is_array() && value.as_array().map_or(false, |a| a.is_empty()))
+                    || (value.is_string() && value.as_str().unwrap().is_empty())
+                    || (value.is_number() && value.as_i64() == Some(0))
+                {
+                    return false;
+                }
+            } else {
+                return false;
+            }
+        }
+
+        // Don't show new objects if they are fields of an object (top 2 levels only)
+        let segments: Vec<&str> = path.split('/').filter(|s| !s.is_empty()).collect();
+        if segments.len() <= 2 {
+            if let si_frontend_mv_types::component::component_diff::SimplifiedAttributeSource::Value { value } = &source_and_value.source.simplified_source {
+                if value.is_object() {
+                    return false;
+                }
+            }
+        }
+    }
```

Comment:
I can't remember. Does the frontend or backend do this now for the regular review screen? I ask only because i'm curious if this logic exists elsewhere via copy/paste in Rust or is brand new for this PR. My end goal is to confirm that we don't have a drift in the subtleties of review between the two location (though, some variance and differences are probably expected and even preferable, depending on what you want to see).